test/functional/api/cas/installer.py

```diff
@@ -8,19 +8,46 @@
 
 from tests import conftest
 from core.test_run import TestRun
+from api.cas.version import CasVersion
 from api.cas import cas_module
 from test_utils import os_utils
 from test_utils.output import CmdException
 
 
-def install_opencas():
+def rsync_opencas_sources():
     TestRun.LOGGER.info("Copying Open CAS repository to DUT")
     TestRun.executor.rsync_to(
         f"{TestRun.usr.repo_dir}/",
         f"{TestRun.usr.working_dir}/",
         exclude_list=["test/functional/results/"],
         delete=True)
 
+
+def _clean_opencas_repo():
+    TestRun.LOGGER.info("Cleaning Open CAS repo")
+    output = TestRun.executor.run(
+        f"cd {TestRun.usr.working_dir} && "
+        "make distclean")
+    if output.exit_code != 0:
+        raise CmdException("make distclean command executed with nonzero status", output)
+
+
+def _checkout_version(version: CasVersion):
```

Comment:
it would make sense not to limit this to tags - checking out by hash will be necessary to test pre-release versions